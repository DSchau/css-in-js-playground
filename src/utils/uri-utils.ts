/*
 * This code lifted from bvaughn/babel-repl
 * Thank you!
 */
import * as LZString from 'lz-string';

export const compress = (string: string) =>
  LZString.compressToBase64(string)
    .replace(/\+/g, '-') // Convert '+' to '-'
    .replace(/\//g, '_') // Convert '/' to '_'
    .replace(/=+$/, ''); // Remove ending '='

export const decompress = (string: string) =>
  LZString.decompressFromBase64(
    string
      .replace(/-/g, '+') // Convert '-' to '+'
      .replace(/_/g, '/') // Convert '_' to '/'
  );
