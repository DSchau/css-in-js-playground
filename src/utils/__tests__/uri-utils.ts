import { compress, decompress } from '../uri-utils';

const snippet = `
import React, { Component } from 'react';
import styled from 'styled-components';

const Heading = styled.h1\`
  color: red;
\`;

export default class Login extends Component {
  render() {
    return <Heading>Hello World</Heading>;
  }
}
`;

test('it compresses code snippet', () => {
  expect(compress(snippet)).toMatchSnapshot();
});

test('it decompresses code snippet', () => {
  const compressed = compress(snippet);

  expect(decompress(compressed)).toBe(snippet);
});
