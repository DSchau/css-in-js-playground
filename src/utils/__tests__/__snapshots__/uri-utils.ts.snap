// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it compresses code snippet 1`] = `"FASwtgDg9gTgLgAgEoFMCGBjOAaBBvBAYSkigDsUzEBfBAMxhIQHIZ0tmBuUU-BAZzgBPADYoAJvUZgWg0RIC0GEtApV-XYMGVlBCABLpxIMgHMEAXgHCx4gHQALAIwADYAgTKRsAFwI24twu3MAoAB7QfOIodGgAriKIGCJo_PwIADJQpiYI4XCU4unEpGqIeO7-hSgwABQAlPiVHmxwcTBkCAA8hmjGZgB8hiLeCADqsCLiXQD0vf2mA9we1MCrQA"`;
